Rebuiling redis image with rpm packages

vishnuchalla opened the pull request:
## Description

Rebuilding redis with rpm packages.

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Closes # https://issues.redhat.com/browse/OLS-312

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
Tested integrated with this PR: https://github.com/openshift/lightspeed-operator/pull/23

vishnuchalla commented:
/assign @williamcaban

bparees commented on Containerfile.redis:
```diff
@@ -1,33 +1,17 @@
-    microdnf clean all
+# Install necessary dependencies for downloading and installing Redis
+RUN microdnf install -y wget yum-utils logrotate systemd && \
+    wget https://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/Packages/redis-7.0.12-1.module_el9+499+353c08ce.x86_64.rpm && \
```
better.  wish we could just dnf install the rpm from the repo directly but it doesn't seem to show up for some reason.

bparees replied:
(even when i tried using a centos streams base image and dnf installing into that).  maybe @williamcaban has an idea

bparees commented:
/approve

syedriko commented on Containerfile.redis:
```diff
@@ -1,33 +1,17 @@
-RUN microdnf install -y gcc make wget tar gzip && \
-    microdnf clean all
+# Install necessary dependencies for downloading and installing Redis
+RUN microdnf install -y wget yum-utils logrotate systemd && \
```
Why are we installing logrotate? Redis logs should be going into stdout/stderr.

vishnuchalla replied:
Yes. But redis itself requires logrotate as a dependency. If we try to build this image without it, we will run into this error
```
error: Failed dependencies:
	logrotate is needed by redis-6.2.7-1.el9.x86_64
```

syedriko commented on Containerfile.redis:
```diff
@@ -1,33 +1,17 @@
-RUN sed -i 's/^protected-mode yes/protected-mode no/' redis.conf
-RUN sed -i 's/^bind 127.0.0.1 -::1/bind 0.0.0.0/' redis.conf
+# Grant necessary permissions
+RUN chmod -R 777 /var/lib/redis /var/log/redis /etc/redis
```
It would be better to move these to the RUN command above so as not to create an extra image layer. Also, the perms seem too permissive. On the cluster, with the restricted-v2 SCC uid is whatever but gid is 0, would 770 be enough?

tisnik replied:
`/etc/redis` might be just `+r`, not `+w` nor `+x`

vishnuchalla replied:
Acknowledged. Combined these with the above RUN command and granted only the required perms.

syedriko commented on Containerfile.redis:
```diff
@@ -1,33 +1,17 @@
+RUN microdnf install -y wget yum-utils logrotate systemd && \
+    wget https://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/Packages/redis-7.0.12-1.module_el9+499+353c08ce.x86_64.rpm && \
+    rpm -i redis-7.0.12-1.module_el9+499+353c08ce.x86_64.rpm && \
+    rm -f redis-7.0.12-1.module_el9+499+353c08ce.x86_64.rpm
```
Could uninstall wget here.

vishnuchalla replied:
Added cleanup commands too. Thanks

bparees commented on Containerfile.redis:
```diff
@@ -1,39 +1,18 @@
+RUN microdnf install -y wget yum-utils logrotate systemd && \
+    wget https://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/Packages/redis-6.2.7-1.el9.x86_64.rpm && \
+    rpm -i redis-6.2.7-1.el9.x86_64.rpm && \
+    rm -f redis-6.2.7-1.el9.x86_64.rpm && \
```
if we're deliberately switching to redis6 (and i'm not sure why we would, since rhel offers redis 7 and v7 will be in support longer), it can definitely be installed on centos using a centos base image, with just a dnf install.

which we should probably doing regardless (using a centos base image, if we're going to add centos content..i don't see a good reason not to):

https://quay.io/repository/centos/centos?tab=tags

syedriko commented:
/lgtm

bparees commented:
/hold

i want to talk about why we're switching to redis6

bparees commented:
/approve cancel

vishnuchalla commented:
> /hold
> 
> i want to talk about why we're switching to redis6

redis 7 disappeared all of a sudden from the official centos stream dowloads: https://pkgs.org/download/redis (i.e. which makes me think if its reliable). So had to switch to redis 6 which also has TLS enabled. But if we are okay with building from quay's centos, I can take a look at it.

bparees commented on Containerfile.redis:
```diff
@@ -1,39 +1,21 @@
-RUN sed -i 's/^# tls-cert-file redis.crt/tls-cert-file \/etc\/certs\/tls.crt/' redis.conf
-RUN sed -i 's/^# tls-key-file redis.key/tls-key-file \/etc\/certs\/tls.key/' redis.conf
-RUN sed -i 's/^# tls-ca-cert-file ca.crt/tls-ca-cert-file \/etc\/certs\/cm-olsredisca\/service-ca.crt/' redis.conf
-RUN sed -i 's/^# tls-auth-clients optional/tls-auth-clients optional/' redis.conf
```
you don't need any of these config file tweaks any more?

vishnuchalla replied:
Those do not look good here, So I am passing them through cli [command](https://github.com/openshift/lightspeed-operator/pull/25/files#diff-0bba25c929831058ca39ea8a7599b3ee823c1eee0798ae2a1a1228665fd33e7dR53).

bparees commented:
/lgtm
/approve

bparees commented:
@vishnuchalla i think this makes our redis containerfile better, but:

1) the commits say it uses rpms, but that's no longer true so please fix the commit text (i already fixed the PR title)
2) this does not actually implement the "install from an rpm" acceptance criteria of OLS-312, so i think we need to put OLS-312 back into our backlog until we resolve the issue w/ getting the centos rpm.

vishnuchalla commented:
> @vishnuchalla i think this makes our redis containerfile better, but:
> 
> 1. the commits say it uses rpms, but that's no longer true so please fix the commit text (i already fixed the PR title)
> 2. this does not actually implement the "install from an rpm" acceptance criteria of [OLS-312](https://issues.redhat.com//browse/OLS-312), so i think we need to put [OLS-312](https://issues.redhat.com//browse/OLS-312) back into our backlog until we resolve the issue w/ getting the centos rpm.

Done. Updated as suggested.

bparees commented:
/hold cancel
/approve
/lgtm
/label tide/merge-method-squash
